Assets/BossRoom/Scripts/Client/UI/Lobby/CreateLobbyUI.cs

```diff
@@ -0,0 +1,140 @@
+        [SerializeField] private CanvasGroup m_CanvasGroup;
+
+        private LobbyUIMediator m_LobbyUIMediator;
+
+        private OnlineMode m_OnlineMode;
+
+        private void Awake()
+        {
+            EnableIPHostUI();
+            m_IPToggle.onValueChanged.AddListener(IPRadioRadioButtonPressed);
+            m_UnityRelayToggle.onValueChanged.AddListener(UnityRelayRadioRadioButtonPressed);
+        }
+
+        [Inject]
+        private void InjectDependencies(LobbyUIMediator lobbyUIMediator)
+        {
+            m_LobbyUIMediator = lobbyUIMediator;
+        }
+
+        private void IPRadioRadioButtonPressed(bool value)
+        {
+            if (!value)
+            {
+                return;
+            }
+
+            EnableIPHostUI();
+        }
+
+        private void UnityRelayRadioRadioButtonPressed(bool value)
+        {
+            if (!value)
+            {
+                return;
+            }
+
+            EnableUnityRelayUI();
+        }
+
+        private void EnableUnityRelayUI()
+        {
+            m_IPConnectionCanvasGroup.alpha = 0;
+            m_IPConnectionCanvasGroup.blocksRaycasts = false;
+            m_IPConnectionCanvasGroup.interactable = false;
+
+            m_LoadingImage.SetActive(false);
+        }
+
+        private void EnableIPHostUI()
+        {
+            m_IPConnectionCanvasGroup.alpha = 1;
+            m_IPConnectionCanvasGroup.blocksRaycasts = true;
+            m_IPConnectionCanvasGroup.interactable = true;
+
+            m_PortInputField.text = k_DefaultPort.ToString();
+            m_PortInputField.text = k_DefaultPort.ToString();
+
+            m_LoadingImage.SetActive(false);
+        }
+
+
+        public void OnCreateClick()
+        {
+            var portNum = 0;
+            int.TryParse(m_PortInputField.text, out portNum);
+            if (portNum <= 0)
+            {
+                portNum = k_DefaultPort;
+            }
+
+            var ip = string.IsNullOrEmpty(m_IPInputField.text) ? k_DefaultIP : m_IPInputField.text;
+
+            m_LobbyUIMediator.CreateLobbyRequest(m_LobbyNameInputField.text, m_IsPrivate.isOn, 8, m_OnlineMode, ip, portNum);
+        }
+
+        public void Show()
+        {
+            m_CanvasGroup.alpha = 1f;
+            m_CanvasGroup.blocksRaycasts = true;
+        }
+
+        public void Hide()
+        {
+            m_CanvasGroup.alpha = 0f;
+            m_CanvasGroup.blocksRaycasts = false;
+        }
+
+
+        /// <summary>
+        /// Sanitize user port InputField box allowing only alphanumerics, plus any matching chars, if provided.
+        /// </summary>
+        /// <param name="dirtyString"> string to sanitize. </param>
+        /// <param name="includeChars"> Array of chars to include. </param>
+        /// <returns> Sanitized text string. </returns>
+        private static string Sanitize(string dirtyString)
+        {
+            return Regex.Replace(dirtyString, "[A-Za-z0-9]", "");
```

Comment:
Whoops, I accidentally suggested the opposite of what you want.  Should be `"[^A-Za-z0-9]"`.  Replace everything OTHER than these characters with nothing. :D